sphinx/directives/other.py

```diff
@@ -39,6 +39,7 @@ class TocTree(Directive):
     final_argument_whitespace = False
     option_spec = {
         'maxdepth': int,
+        'name': str,
```

Comment:
That makes sense. In my testing though, it was more backwards incompatible. For instance, I am getting weird parse errors when you have 4-line indented options for the toctree:

```
.. toctree:: <spaces here>
    :maxdepth: 2
```

I like the explicitness of a name option because it doesn't heavily change how things have been parsed historically. It's likely that I might just be failing at writing it properly though, and could be convinced otherwise.